src/cpp/sensitivity/SensitivityLogger.cpp

```diff
@@ -14,14 +14,47 @@ void SensitivityLogger::display_message(const std::string& msg) {
   _stream << msg << std::endl << std::endl;
 }
 
-void SensitivityLogger::log_at_start(const SensitivityOutputData& output_data) {
+void SensitivityLogger::log_at_start(const SensitivityInputData& input_data) {
   _stream << std::endl;
-  _stream << "Best overall cost = "
-          << xpansion::logger::commons::create_str_million_euros(
-                 output_data.best_benders_cost)
-          << MILLON_EUROS << std::endl;
-  _stream << "epsilon = " << output_data.epsilon << EUROS << std::endl;
+  _stream << "--- Recall of the best investment solution ---" << std::endl
+          << std::endl;
+
+  log_benders_overall_cost(input_data.best_ub);
+  log_benders_capex(input_data.benders_capex);
+  log_benders_solution(input_data.benders_solution);
+  log_epsilon(input_data.epsilon);
+
+  _stream << std::endl;
+}
+
+void SensitivityLogger::log_benders_overall_cost(const double& best_ub) {
+  _stream << indent_1 << "Best overall cost = "
+          << xpansion::logger::commons::create_str_million_euros(best_ub)
+          << MILLON_EUROS << std::endl
+          << std::endl;
+}
+
+void SensitivityLogger::log_benders_capex(const double& benders_capex) {
+  _stream << indent_1 << "Best capex = "
+          << xpansion::logger::commons::create_str_million_euros(benders_capex)
+          << MILLON_EUROS << std::endl
+          << std::endl;
+}
+
+void SensitivityLogger::log_benders_solution(
+    const std::map<std::string, double>& benders_solution) {
+  _stream << indent_1 << "Best investment solution = " << std::endl;
+  for (const auto& kvp : benders_solution) {
```

Comment:
You can use structured binding. You can even use better names than "key" and "value", like "candidate" and "investement" or what is proper in this case.
```suggestion
  for (const auto& [key, value] : benders_solution) {
      _stream << indent_1 << indent_1 << key << ": "
            << xpansion::logger::commons::create_str_mw(value) << MW
```

Reply:
Done